src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/LocationAttribute.kt

```diff
@@ -144,11 +129,18 @@ class LocationAttribute private constructor(
   override fun toString(): String = this::class.simpleName +
     "(locationAttributeId = $locationAttributeId, prisonId = ${prison.prisonId}, dpsLocationId = $dpsLocationId)"
 
-  fun isAvailableFor(probationTeam: ProbationTeam, onDate: LocalDate, startTime: LocalTime, endTime: LocalTime): AvailabilityStatus = check(probationTeam, onDate, startTime, endTime)
-
-  private fun check(probationTeam: ProbationTeam, onDate: LocalDate, startTime: LocalTime, endTime: LocalTime): AvailabilityStatus {
+  fun isAvailableFor(usageType: LocationUsageType, onDate: LocalDate, startTime: LocalTime, endTime: LocalTime): AvailabilityStatus = run {
     if (locationStatus == LocationStatus.INACTIVE) return AvailabilityStatus.NONE
 
+    if (locationStatus == LocationStatus.TEMPORARILY_BLOCKED && onDate.between(blockedFrom!!, blockedTo)) return AvailabilityStatus.NONE
+
+    when (usageType) {
+      is Court -> check(usageType, onDate, startTime, endTime)
```

Comment:
Should there be a Shared usage type here? Or is usageType the type of booking?

Reply:
It's just a naming thing.  I'll try and think of a better name 🤔